app/gateway/database/postgres/account/create.go

```diff
@@ -3,14 +3,13 @@ package account
 import (
 	"context"
 	"stonehenge/app/core/entities/account"
-	"stonehenge/app/core/types/id"
 	"stonehenge/app/gateway/database/postgres/common"
 )
 
-func (r *repository) Create(ctx context.Context, acc *account.Account) (id.ExternalID, error) {
+func (r *repository) Create(ctx context.Context, acc account.Account) (account.Account, error) {
 	db, found := common.TransactionFrom(ctx)
 	if !found {
-		return id.New(), account.ErrCreating
+		return acc, account.ErrCreating
```

Comment:
We usually return an empty struct in our projects.

```suggestion
		return account.Account{}, account.ErrCreating
```

Reply:
Also, if you return a custom error, whoever is using the function is going to miss valuable information.
You may just return the error the database returned and let the other layers deal with it.

Reply:
Oh, so it's ok to let a DB error leave its layer, for example?